dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -1538,6 +1545,63 @@ def test_save_and_load_no_labeler(self):
         save_profile.update_profile(pd.DataFrame({"a": [4, 5]}))
         load_profile.update_profile(pd.DataFrame({"a": [4, 5]}))
 
+    def test_save_json_file(self):
+        self.maxDiff = None
+        datapth = "dataprofiler/tests/data/"
+        test_file = "csv/iris.csv"
+
+        # Create Data and StructuredProfiler objects
+        data = dp.Data(os.path.join(datapth, test_file))
+        options = ProfilerOptions()
+        options.set({"correlation.is_enabled": True})
+        with test_utils.mock_timeit():
+            save_profile = dp.StructuredProfiler(data)
+
+        # Save and Load profile with Mock IO
+        with mock.patch("builtins.open") as m:
+            mock_file = setup_save_mock_string_open(m)
+            save_profile.save("mock.json")
+            mock_file.seek(0)
```

Comment:
two thoughts (not hard and fast here), but we should validate the return_value from `json.dumps()` here... I say that because the error in the code above would have been caught. Maybe there is a return_value from `json.dumps()` that we could validate the `filepath` variable that is being actually written... we'd have to mock I reckon

Reply:
or just add a test validating passing `None` to the `filepath` parameter... and validate it outputs the autogenerate filepath (would have to mock the time generation)